actors/util/smoothing/alpha_beta_filter.go

```diff
@@ -175,18 +175,21 @@ func ExtrapolatedCumSumOfRatio(delta abi.ChainEpoch, relativeStart abi.ChainEpoc
 }
 
 // The natural log of Q.128 x.
-func ln(z big.Int) big.Int {
-	// ln(z) = ln(x * 2^k) = ln(x) + k * ln2
-	intK := uint(z.BitLen() - 1 - math.Precision)
-	k := big.NewInt(int64(intK))   // Q.0
+func Ln(z big.Int) big.Int {
+	// bitlen - 1 - precision
+	k := big.Sub(big.Sub(big.NewInt(int64(z.BitLen())), big.NewInt(1)), big.NewInt(math.Precision)) // Q.0
```

Comment:
What is the case when you can't do this math in uint/64 precison?

Reply:
If you are referring to the k calculation only we should always be able to.  I switched to big int after doing this math with uints and missing the sign on k when switching k to a big.  I can switch to int64.  Is the suggestion for performance?